# Priorización de clientes para campaña de depósitos a plazo

## Pregunta de negocio

¿Qué segmentos de clientes y qué canales deben priorizarse para maximizar la contratación de depósitos a plazo sin aumentar el volumen de contactos?

> Este análisis es descriptivo. Para evitar fuga de información, se excluyen `duration` y `campaign`, porque no están disponibles al decidir el primer contacto.

In [1]:
import pandas as pd

from bank_customer.extract.extract import extract
from bank_customer.transform.clean_df import clean_heads
from bank_customer.utils.paths import get_path_raw

df = clean_heads(extract(get_path_raw("dataset.csv")))
df["converted"] = df["subscription"].eq(2)

df.head()

,age,type_of_job,marital_status,education_level,credit_in_default,average_yearly_balance,housing_loan,personal_loan,contact_communication_type,last_contact_day_of_the_month,month,duration,campaign,pdays,previous,poutcome,subscription,converted
0,58,management,married,tertiary,no,2143,yes,no,unknown,5,may,261,1,-1,0,unknown,1,False
1,44,technician,single,secondary,no,29,yes,no,unknown,5,may,151,1,-1,0,unknown,1,False
2,33,entrepreneur,married,secondary,no,2,yes,yes,unknown,5,may,76,1,-1,0,unknown,1,False
3,47,blue-collar,married,unknown,no,1506,yes,no,unknown,5,may,92,1,-1,0,unknown,1,False
4,33,unknown,single,unknown,no,1,no,no,unknown,5,may,198,1,-1,0,unknown,1,False


## Contexto general

Se calcula la conversión global. Esta será la referencia para evaluar si un segmento está por encima o por debajo del promedio.

In [2]:
overall_conversion_rate = df["converted"].mean()

overall_summary = pd.DataFrame(
    {
        "customers": [len(df)],
        "conversions": [df["converted"].sum()],
        "conversion_rate_pct": [overall_conversion_rate * 100],
    }
).round(1)

overall_summary

,customers,conversions,conversion_rate_pct
0,45211,5289,11.7


In [3]:
def conversion_summary(data: pd.DataFrame, group_columns: str | list[str]) -> pd.DataFrame:
    summary = (
        data.groupby(group_columns, observed=True)
        .agg(
            customers=("subscription", "size"),
            conversions=("converted", "sum"),
            conversion_rate=("converted", "mean"),
        )
        .reset_index()
    )

    summary["conversion_rate_pct"] = (summary["conversion_rate"] * 100).round(1)
    summary["uplift_vs_global_pp"] = ((summary["conversion_rate"] - overall_conversion_rate) * 100).round(1)

    return summary.sort_values(["conversion_rate", "customers"], ascending=[False, False]).drop(columns="conversion_rate")

## Conversión por canal de contacto

In [4]:
channel_summary = conversion_summary(df, "contact_communication_type")
channel_summary

,contact_communication_type,customers,conversions,conversion_rate_pct,uplift_vs_global_pp
0,cellular,29285,4369,14.9,3.2
1,telephone,2906,390,13.4,1.7
2,unknown,13020,530,4.1,-7.6


## Conversión por perfil de cliente

In [5]:
df["age_group"] = pd.cut(
    df["age"],
    bins=[17, 30, 40, 50, 60, float("inf")],
    labels=["18-29", "30-39", "40-49", "50-59", "60+"],
    right=False,
)

In [6]:
age_summary = conversion_summary(df, "age_group")
age_summary

,age_group,customers,conversions,conversion_rate_pct,uplift_vs_global_pp
4,60+,1784,600,33.6,21.9
0,18-29,5273,928,17.6,5.9
1,30-39,18089,1913,10.6,-1.1
3,50-59,8410,785,9.3,-2.4
2,40-49,11655,1063,9.1,-2.6


In [7]:
housing_summary = conversion_summary(df, "housing_loan")
housing_summary

,housing_loan,customers,conversions,conversion_rate_pct,uplift_vs_global_pp
0,no,20081,3354,16.7,5.0
1,yes,25130,1935,7.7,-4.0


In [8]:
previous_campaign_summary = conversion_summary(df, "poutcome")
previous_campaign_summary

,poutcome,customers,conversions,conversion_rate_pct,uplift_vs_global_pp
2,success,1511,978,64.7,53.0
1,other,1840,307,16.7,5.0
0,failure,4901,618,12.6,0.9
3,unknown,36959,3386,9.2,-2.5


## Conclusiones y recomendación

La tasa de conversión global fue de **11.7 %** (5,289 suscripciones de 45,211 clientes).

- **Priorizar contactos por celular:** este canal alcanzó una conversión de **14.9 %**, 3.2 puntos porcentuales por encima del promedio, y concentra el mayor volumen de conversiones (**4,369**).
- **Priorizar clientes con campaña previa exitosa:** el segmento `poutcome = success` tuvo una conversión de **64.7 %** (978 de 1,511 clientes), 53.0 puntos porcentuales sobre la tasa global. Es la señal más fuerte para priorización, aunque debe considerarse su menor tamaño relativo.
- **Considerar perfiles con mayor propensión:** los clientes de **60+** alcanzaron **33.6 %** de conversión y quienes no tienen préstamo hipotecario alcanzaron **16.7 %**. Estos segmentos pueden complementar la priorización basada en el resultado de campañas previas.
- **Revisar registros con canal desconocido:** el grupo `unknown` tuvo solo **4.1 %** de conversión. Antes de descartarlo por completo, conviene revisar si representa falta de calidad de datos o una condición operativa distinta.

Estos hallazgos son descriptivos y no demuestran causalidad. Para evitar fuga de información, se excluyeron `duration` y `campaign`, ya que no están disponibles al decidir el primer contacto.